### PR TITLE
added provider label 

### LINKED ISSUE #212: Add clusterctl provider label to BYOH infrastructure components
`Clusterctl` integration requires provider label(`clusterctl.cluster.x-k8s.io/provider`) to be available with all `BYOH` infrastructure components. Details [here](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#labels).

This label `cluster.x-k8s.io/provider: "infrastructure-byoh"` has to be added here in [kustomization.yaml](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/config/default/kustomization.yaml#L12) and validate the same via `make publish-infra-yaml`.

### PR DESCRIPTION
**Pull request by himanshu007-creator**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #212

**Additional information**

**Special notes for your reviewer**

### CODE REVIEW

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ namePrefix: byoh-
 
 # Labels to add to all resources and selectors.
 #commonLabels:
-#  someName: someValue
+cluster.x-k8s.io/provider: "infrastructure-byoh"
```
Comment (dharmjit) [inline config/default/kustomization.yaml:13]: We need to uncomment commonLabels and it should be like below
```
commonLabels:
  cluster.x-k8s.io/provider: "infrastructure-byoh"
```

**Comment by anusha94** [comment]
Hey @himanshu007-creator , Do you need any help in progressing with this PR?

File: config/default/kustomization.yaml (hunk 1 above)
Comment (himanshu007-creator) [reply config/default/kustomization.yaml:13]: Okay👍🏽

**Comment by himanshu007-creator** [comment]
> Hey @himanshu007-creator , Do you need any help in progressing with this PR?

Thanks, @anusha94 🌝, but I have made the changes suggested by @dharmjit.

**Review by dharmjit — APPROVED** [review]
Thanks, @himanshu007-creator for your contribution. LGTM!

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**